### PR TITLE
refactor: shape perf

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by Xiaoy312** [comment]
fixme@xy: 
- [ ] package diff ignore

File: src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs [hunk 1]
```diff
@@ -203,7 +241,35 @@ private UIElementCollection GetDefs()
...
-			return shape._mainSvgElement.GetBBox();
+			if (FeatureConfiguration.Shape.CacheBBoxCalculationResult)
+			{
+				var key = shape switch
+				{
+					// most other properties don't impact getBBox result, see also comments in UpdateRender.
+					Path path => $"path:{(path.Data as GeometryData)?.Data}",
+					Line line => $"line:{line.X1},{line.Y1}-{line.X2}{line.Y2}",
+					Polygon polygon => $"polygon:{polygon.Points?.ToCssString()}",
+					Polyline polyline => $"polyline:{polyline.Points?.ToCssString()}",
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs:252]: I wonder, should we cache this further? Maybe by computing those strings ahead of time?

File: src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs [hunk 2]
```diff
@@ -1,27 +1,45 @@
...
-using System.Numerics;
-using System.Diagnostics;
-using Uno.UI.Xaml;
 
 namespace Microsoft.UI.Xaml.Shapes
 {
+	partial class Shape
+
+	{
+		private static readonly LruCache<string, Rect> _bboxCache = new(500);
```
Comment (Xiaoy312) [inline src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs:24]: this default cache of 500 make senses?

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs:24]: Yes, that should be enough for now. We're not caching much, so it should not have a great memory impact.

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.5

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.4

**Pull request merged**